sql/hive/src/test/scala/org/apache/spark/sql/hive/orc/HiveOrcQuerySuite.scala

```diff
@@ -313,4 +314,71 @@ class HiveOrcQuerySuite extends OrcQueryTest with TestHiveSingleton {
       }
     }
   }
+
+  test("SPARK-32864: Support ORC forced positional evolution") {
+    Seq("native", "hive").foreach { orcImpl =>
+      Seq(true, false).foreach { forcePositionalEvolution =>
+        withSQLConf(SQLConf.ORC_IMPLEMENTATION.key -> orcImpl,
+          OrcConf.FORCE_POSITIONAL_EVOLUTION.getAttribute -> forcePositionalEvolution.toString) {
+          withTempPath { f =>
+            val path = f.getCanonicalPath
+            Seq((1, 2), (3, 4), (5, 6)).toDF("c1", "c2").write.orc(path)
+            val correctAnswer = Seq(Row(1, 2), Row(3, 4), Row(5, 6))
+            checkAnswer(spark.read.orc(path), correctAnswer)
+
+            withSQLConf(HiveUtils.CONVERT_METASTORE_ORC.key -> "true") {
```

Comment:
Can we a test coverage for `false`?

Reply:
Sure, added in https://github.com/apache/spark/pull/29737/commits/51f503c5738a714d6ea77467ac8f7dfba231d989